javalib/src/main/scala/java/lang/System.scala

```diff
@@ -82,4 +82,31 @@ object System {
       stdio.fwrite(buf, 1, 1, stream)
     }
   }
+
+  private class CStdinStream() extends java.io.InputStream {
+    def read(): CInt = {
+      val buffer = stackalloc[scala.Byte](1)
+      var nread  = unistd.read(0, buffer, 1)
+      if (nread < 1)
+        -1
```

Comment:
If read is `-1` the value of `errno` should be checked. In case it is `EAGAIN` or `EINTR` it is safe to retry the read, else an error should probably be thrown.